### PR TITLE
Added basic measure tx and measure rx capabilities to check the usage…

### PR DESCRIPTION
**Pull request by jjtbsomhorst**
Added basic measure tx/rx capabilities that show up on the different devices.I have to polish it a bit ( so it shows Mb, Gb etc..) but it's a funny addition

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -31,11 +31,11 @@ class UnifiApp extends Homey.App {
 
         this.log('Freshly initializing com.ubnt.unifi.settings with some defaults')
         this.updateSettings({
-            'host': 'unifi',
-            'port': '8443',
-            'user': 'ubnt',
-            'pass': 'ubnt',
-            'site': 'default'
+            'host': '',
```
Comment (mnederlof) [inline app.js:34]: Why did you remove the defaults?

File: app.json [hunk 2]
```diff
@@ -68,6 +68,26 @@
             "min": -100,
             "max": 0,
             "step": 1
+        },
+        "measure_bytes_send":{
```
Comment (mnederlof) [inline app.json:72]: I think this should be named measure_bytes_sent (past tense), to make it in correct english grammar.

File: drivers/wifi-client/driver.js [hunk 3]
```diff
@@ -67,6 +68,7 @@ class UnifiDriver extends Homey.Driver {
         // One poller for initializing the connection every 60 seconds (if needed)
         this.pollIntervals.push(setInterval(() => {
             this.initializeConnection();
+            this.getUsageData();
```
Comment (mnederlof) [inline drivers/wifi-client/driver.js:71]: Why do you fetch usage data here? Shouldn't it be part of the updateClientList and updating the tx/rx values be part of the sendDeviceUpdates method (e.g. be part of drivers/wifi-client/device.js and be updated there)?

File: drivers/wifi-client/driver.js [hunk 4]
```diff
@@ -84,6 +86,56 @@ class UnifiDriver extends Homey.Driver {
...
+                    let unifyData = mappedData.get(deviceData.id);
+                    if(unifyData.hasOwnProperty('tx_bytes')){
+                        tx = unifyData.tx_bytes / GB_DIVISOR;
+                    }
+                    if(unifyData.hasOwnProperty('rx_bytes')){
+                        rx = unifyData.rx_bytes / GB_DIVISOR;
+                    }
+                }
+
+                device.setCapabilityValue("measure_bytes_received",tx,(err,result)=>{
```
Comment (mnederlof) [inline drivers/wifi-client/driver.js:118]: Updating values should be done through the device driver in the unifi app. Preferably by either creating a new method in the device driver, or by incorporating already existing methods (and at least by by using private _updateProperty method, which is updating only if data has changed.)

File: drivers/wifi-client/driver.js [hunk 5]
```diff
@@ -576,7 +628,7 @@ class UnifiDriver extends Homey.Driver {
 
                     let name = client.name
                     if (typeof name === 'undefined') name = client.hostname;
-
+                    if (typeof name === 'undefined') name = client.mac;
```
Comment (mnederlof) [inline drivers/wifi-client/driver.js:631]: You had issues with clients having no hostname ? 😄 (i've not seen this sofar.. :) )

**Review by mnederlof — CHANGES_REQUESTED** [review]

File: app.js (hunk 1 above)
Comment (jjtbsomhorst) [reply app.js:34]: Not really necessary but liked it a bit more while testing

File: drivers/wifi-client/driver.js (hunk 5 above)
Comment (jjtbsomhorst) [reply drivers/wifi-client/driver.js:631]: Yeah have several devices that for some reason don't send their hostname..
